### PR TITLE
WIP: COG-14

### PR DESCRIPTION
**Pull request by amanjeev**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -317,6 +317,10 @@ clean-volumes:
 compose:
 	$(foreach MODULE, $(CANDIG_MODULES), $(MAKE) compose-$(MODULE);)
 
+
+compose-kc:
```
Comment (shaikh-rashid) [inline Makefile:321]: this is not needed as `Makefile` contains a method for calling individual modules from `lib/`

File: Makefile [hunk 2]
```diff
@@ -73,7 +73,7 @@ bin-docker-machine: mkdir
 bin-kompose: mkdir
 	curl -Lo $(DIR)/bin/kompose \
 		https://github.com/kubernetes/kompose/releases/download/v1.21.0/kompose-$(VENV_OS)-amd64
-	chmod 755 $(DIR)/bin/kubectl
+	chmod 755 $(DIR)/bin/kompose
```
Comment (shaikh-rashid) [inline Makefile:76]: already addressed in branch `develop`. This will be merged into stable in another PR

File: lib/keycloak/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+version: '3.7'
```
Comment (shaikh-rashid) [inline lib/keycloak/docker-compose.yml:1]: this is a good start, please see `lib/templates/docker-compose.yml` for a reference `docker-compose` file

File: lib/keycloak/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    container_name: keycloak
+    command: ["-b", "0.0.0.0", "-Dkeycloak.migration.strategy=IGNORE_EXISTING"]
+    ports:
+      - "${KC_LOCAL_PORT}:8081"
+    env_file:
+      - ./volumes/secrets.env
+    #networks:
+    #  - ${DOCKER_NET}
+    volumes:
+      - ./volumes/keycloak-db:/opt/jboss/keycloak/standalone/data
```
Comment (shaikh-rashid) [inline lib/keycloak/docker-compose.yml:30]: please nest volumes in `tmp/data` when testing, or create volumes via `docker volume create`. `Makefile` also contains a method for creating docker volumes in `docker-volumes` section. remember that local directories are not scalable so managed volumes/persistent volume claims offer a more flexible solution.

File: lib/keycloak/volumes/application-roles.properties [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/application-roles.properties:1]: these are runtime configs not necessarily volumes. please see https://docs.docker.com/engine/reference/commandline/config/

File: lib/keycloak/volumes/application-users.properties [hunk 6]
```diff
@@ -0,0 +1,25 @@
+#
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/application-users.properties:1]: these are runtime configs not necessarily volumes. please see https://docs.docker.com/engine/reference/commandline/config/

File: lib/keycloak/volumes/logging.properties [hunk 7]
```diff
@@ -0,0 +1,70 @@
+#
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/logging.properties:1]: these are runtime configs not necessarily volumes. please see https://docs.docker.com/engine/reference/commandline/config/

File: lib/keycloak/volumes/mgmt-groups.properties [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/mgmt-groups.properties:1]: these are runtime configs not necessarily volumes. please see https://docs.docker.com/engine/reference/commandline/config/

File: lib/keycloak/volumes/mgmt-users.properties [hunk 9]
```diff
@@ -0,0 +1,27 @@
+#
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/mgmt-users.properties:1]: these are runtime configs not necessarily volumes. please see https://docs.docker.com/engine/reference/commandline/config/

File: lib/keycloak/volumes/secrets.env [hunk 10]
```diff
@@ -0,0 +1,3 @@
+KEYCLOAK_USER=kcadmin
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/secrets.env:1]: storing secrets in a repo file is not recommended, please see https://docs.docker.com/engine/swarm/secrets/

File: lib/keycloak/volumes/secrets.env [hunk 11]
```diff
@@ -0,0 +1,3 @@
...
+KEYCLOAK_PASSWORD=admin
+PROXY_ADDRESS_FORWARDING=false
```
Comment (shaikh-rashid) [inline lib/keycloak/volumes/secrets.env]: It is recommended to store `env` values inside `.env` in the project root directory. Please see `etc/env/example.env`

File: lib/tyk/docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    deploy:
+      placement:
+        constraints:
+          - node.role == manager
+      restart_policy:
+        condition: on-failure
+        delay: 5s
+        max_attempts: 3
+        window: 120s
+    volumes:
```
Comment (shaikh-rashid) [inline lib/tyk/docker-compose.yml:24]: This strategy of volume configuration is not recommended, instead, consider creating a `tyk-gateway` volume with `middleware`, `apps`, and `policies` subdirectories. this way you can have 1 volume mount to `/opt/tyk-gateway`

File: lib/vault/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,45 @@
+# From https://www.bogotobogo.com/DevOps/Docker/Docker-Vault-Consul.php
+# base image
+FROM alpine:3.7
+
+# set vault version
+ENV VAULT_VERSION 1.5.0
```
Comment (shaikh-rashid) [inline lib/vault/Dockerfile:6]: vault version should be a runtime arg instead if possible, see `lib/templates/docker-compose.yml` and `lib/templates/Dockerfile` for examples

File: lib/vault/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+# maintainer
+LABEL maintainer="brennan.brouillette@c3g.ca"
+
+# create a new directory
+RUN mkdir /vault
+RUN mkdir /vault/config
```
Comment (shaikh-rashid) [inline lib/vault/Dockerfile:13]: can just be `mkdir -p /vault/config`

File: lib/vault/docker-compose.yml [hunk 15]
```diff
@@ -0,0 +1,24 @@
+version: '3.6'
```
Comment (shaikh-rashid) [inline lib/vault/docker-compose.yml]: see `lib/templates/docker-compose.yml` and `lib/templates/Dockerfile` for examples

**Comment by shaikh-rashid** [comment]
Overall, this PR is good for PoC but not ready for merge. It is also recommended to open a new PR with this branch and set merge target to `develop` instead of `stable`

**Review by fcoralsasso — DISMISSED** [review]
lgtm

File: Makefile (hunk 2 above)
Comment (amanjeev) [reply Makefile:76]: @shaikh-rashid : I don't think this is addressed in `develop` yet: 

https://github.com/CanDIG/CanDIGv2/blob/develop/Makefile#L73-L76

**Comment by amanjeev** [comment]
Closed because stale work. New PR will be opened by @brouillette

**Pull request closed without merging**